### DISCUSSION TITLE
My GDAP isnt working without DAP

### DISCUSSION THREAD

**Discussion by elliottabush** [KelvinTegelaar/CIPP#1814] [Q&A]
We recently setup CIPP. We had it working properly using Delegated Access Permissions (DAP). We're migrating to GDAP (granular delegated..) but having trouble.

Here's what's happening to me and i dont understand why...
1. If a tenant has only DAP roles, then CIPP reports that the connection is successful but it's missing GDAP roles
2. If a tenant has only GDAP roles, then CIPP reports it cannot communicate at all.
3. If a tenant has both, then cipp reports all successful.

Does that sound right? I thought that once i migrate to GDAP then DAP roles can be removed. Maybe i'm not understanding it properly.

**Comment by KelvinTegelaar** [comment]
This isnt the place for support, only soonsors get support.
